src/newcharacter.cpp

```diff
@@ -921,9 +921,7 @@ void avatar::initialize( character_type type )
     }
 
     std::vector<matype_id> all_styles = martial_arts_data->get_known_styles( false );
-    int num = rng( 0, all_styles.size() );
-    const matype_id rand_style = all_styles[num];
-    martial_arts_data->set_style( rand_style );
+    martial_arts_data->set_style( random_entry( all_styles ) );
```

Comment:
Maybe this?

```suggestion
    martial_arts_data->set_style( random_entry( all_styles, style_none ) );
```

And then add somewhere at the top

```
static const matype_id style_none( "style_none" );
```

Reply:
The reason being `random_entry` returns a default constructed `V` but the default `matype_id` is not the empty string but `"style_none"`, looking at eg. `character_martial_arts::reset_style()`.